src/arraymancer/tensor/display.nim

```diff
@@ -13,22 +13,160 @@
 # limitations under the License.
 
 import  ./private/p_display,
-        ./data_structure,
-        typetraits
+        ./data_structure
+import std / typetraits
 
-proc pretty*[T](t: Tensor[T], precision = -1): string =
-  ## Pretty-print a Tensor with the option to set a custom `precision`
-  ## for float values.
-  var desc = t.type.name & " of shape \"" & $t.shape & "\" on backend \"" & "Cpu" & "\""
-  if t.storage.isNil: # return useful message for uninit'd tensors instead of crashing
-    return "Uninitialized " & $desc
-  if t.size() == 0:
-    return desc & "\n    [] (empty)"
-  elif t.rank == 1: # for rank 1 we want an indentation, because we have no `|`
-    return desc & "\n    " & t.prettyImpl(precision = precision)
-  else:
-    return desc & "\n" & t.prettyImpl(precision = precision)
+proc pretty*[T](t: Tensor[T], precision: int = -1, showHeader: static bool = true): string =
+  ## Pretty-print a Tensor as a "table" with a given precision and optional header
+  ##
+  ## Pretty-print a Tensor with options to set a custom `precision` for float
+  ## values and to show or hide a header describing the tensor type and shape.
+  ##
+  ## Inputs:
+  ##   - Input Tensor.
+  ##   - precision: The number of decimals printed (for float tensors),
+  ##                _including_ the decimal point.
+  ##   - showHeader: If true (the default) show a dscription header
```

Comment:
```suggestion
  ##   - showHeader: If true (the default) show a description header
```

Reply:
Will fix this typo in a commit taking care of all your comments.